lib/FrontendTool/FrontendTool.cpp

```diff
@@ -122,18 +122,26 @@ static bool emitMakeDependenciesIfNeeded(DiagnosticEngine &diags,
   auto escape = [&](StringRef raw) -> StringRef {
     pathBuf.clear();
 
-    static const char badChars[] = " $#:\n";
-    size_t prev = 0;
-    for (auto index = raw.find_first_of(badChars); index != StringRef::npos;
-         index = raw.find_first_of(badChars, index+1)) {
-      pathBuf.append(raw.slice(prev, index));
-      if (raw[index] == '$')
-        pathBuf.push_back('$');
-      else
+    for (unsigned I = 0, E = raw.size(); I != E; ++I) {
+      switch (raw[I]) {
+      case '#': // Handle '#' the broken GCC way
+        pathBuf.push_back('\\');
+        break;
+
+      case ' ':
+      case '\t':
+        for (unsigned J = I; J && raw[J - 1] == '\\'; --J)
```

Comment:
What if `I` is 0? Also, what is this defending against?

Reply:
`J &&` would be false there, and not enter the loop.

Reply:
Err, yes.  Added a comment explaining whats going on there.